contents/dataframes_performance.md

```diff
@@ -4,8 +4,237 @@ So far, we didn't think about making our `DataFrames.jl` code **fast**.
 Like everything in Julia, `DataFrames.jl` can be really fast.
 In this section, we will give some performance tips and tricks.
 
-```{=comment}
-- Functions with a shebang versus without (e.g., filter! versus filter)
-- df[!, col] versus df[:, col]
-- other things? Jose probably knows some more
+### Allocations
+
+Like we explained in @sec:function_bang functions that end with a bang `!` are a common pattern to denote functions that modify one or more of their arguments.
+In the context of high performance Julia code, this *means* that **functions with `!` do not perform any sort of allocation**.
```

Comment:
I don't that this holds in general. As a counterexample, I can think of a transform operation with some complex logic inside the transformation function.

```
julia> df = DataFrame(A = [1]);

julia> @allocated transform!(df, :A => x -> (rand(100); x))
1168837
```

Reply:
How do you advise a better phrasing?

Reply:
Remove the sentence and only mention the in place part.

Reply:
Can you suggest in the PR so I can see it and apply or edit if necessary?